src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -305,6 +305,16 @@ private static void OnRequestError(WinHttpRequestState state, Interop.WinHttp.WI
             
             Debug.Assert(state != null, "OnRequestError: state is null");
 
+            // Add instrumentation for now to catch unexpected errors and fail fast and produce a crash dump.
+            // Issue #7812
+#if DEBUG
+            if (unchecked((int)asyncResult.dwError) == Interop.WinHttp.ERROR_INSUFFICIENT_BUFFER ||
+                unchecked((int)asyncResult.dwError) == unchecked((int)0x80090321)) // SEC_E_BUFFER_TOO_SMALL
```

Comment:
Error code is different, one is ERROR_INSUFFICIENT_BUFFER, and the other is SEC_E_BUFFER_TOO_SMALL, which is in Interop.SECURITY_STATUES. Those are the two potential error code cause the error message, "buffer too small".